components/feature/search/src/main/java/mozilla/components/feature/search/ext/BrowserStore.kt

```diff
@@ -27,3 +30,28 @@ fun BrowserStore.toDefaultSearchEngineProvider() = object : DefaultSearchEngineP
         return getDefaultSearchEngine()
     }
 }
+
+/**
+ * Waits for the search state to be loaded from disk (when using `RegionMiddleware` and
```

Comment:
nit: Let's point out that this method doesn't block. I think `wait` could be misinterpreted? I had to double-check. Alternative would be to rename the method to `withDefaultSearchEngine` or similar.

Reply:
Good point. Added this to the comment.